### ISSUE TITLE
istio-proxy not intercepting traffic in 1.1.1

### ISSUE THREAD

**Issue by alunsford** [istio/istio#13022]
**Describe the bug**
The istio-proxy sidecar does not intercept in-cluster GRPC traffic in 1.1.1. We have a Service with a port named `grpc` and in 1.0.6, GRPC traffic would get intercepted by the proxy and routed accordingly. In 1.1.1, traffic is not handled by the proxy - it goes straight to the service container running in the pod. We also noticed that another port on the service for health checks, named `http`, was not getting intercepted. We observed EDS sync statuses of 50% for our proxies when running `istioctl proxy-status`. Running a diff of the configs shows that pilot knows about the proxies but envoy does not (https://github.com/istio/istio/issues/10933#issuecomment-476296801).

**Expected behavior**
istio-proxy/pilot should intercept and route traffic for services.

**Steps to reproduce the bug**
- Install istio 1.1.1 on GKE
- Create a deployment with a service, using a named port of `grpc` to route back to the pods
- Configure sidecar injection such that the sidecar is only injected via pod annotation
- Send GRPC traffic from another pod in the same namespace to the service's address (servicename.svc.cluster.local)

**Version**
1.1.1

**Installation**
We downloaded the 1.1.1 release from the releases page and installed via the helm chart. We installed the `istio-init` chart before installing the `istio` chart. After observing the issue that way, we followed the instructions here: https://istio.io/docs/setup/kubernetes/additional-setup/cni/#gke-setup by enabling network policy and installing the CNI addon. Doing so did not change behavior.

**Environment**
- Google Kubernetes Engine (GKE) v1.12.6-gke.7

**Notes**
Created with suggestion from https://github.com/istio/istio/pull/11177#issuecomment-479293194

**Comment by envintus** [comment]
@hzxuzhonghu Please let us know if you need any additional information.

**Comment by hzxuzhonghu** [comment]
you mean the traffic is not intercepted by sidecar? What is you interception mode setting?

**Comment by howardjohn** [comment]
Looks related to CNI @tiswanso maybe can help

**Comment by alunsford** [comment]
@hzxuzhonghu Yes, traffic is not intercepted by the sidecar. Interception is set to REDIRECT. From the envoy admin console for the sidecar:

```
{
   "bootstrap": {
    "node": {
     "id": "sidecar~10.16.35.4~scheduler-6f9b648b65-nb2r8.prill~prill.svc.cluster.local",
     "cluster": "scheduler",
     "metadata": {
      "prometheus.io/scrape": "true",
      "sidecar.istio.io/inject": "true",
      "release.name": "prill-scheduler",
      "prometheus.io/path": "/metrics",
      "prometheus.io/port": "8080",
      "transloc/chart.version": "0.2.1",
      "POD_NAME": "scheduler-6f9b648b65-nb2r8",
      "istio": "sidecar",
      "ISTIO_PROXY_VERSION": "1.0.2",
      "team": "product",
      "ISTIO_PROXY_SHA": "istio-proxy:930841ca88b15365737acb7eddeea6733d4f98b9",
      "transloc/component.version": "20190326-45",
      "component": "scheduler",
      "app": "scheduler",
      "pod-template-hash": "6f9b648b65",
      "transloc/timestamp": "2019-04-03 14:49:46.262481961 +0000 UTC m=+400563.312956810",
      "INTERCEPTION_MODE": "REDIRECT",
      "chart": "scheduler",
      "version": "20190326-45",
      "ISTIO_VERSION": "1.0.6"
     },
```

**Comment by tiswanso** [comment]
It looks like this is occurring with and without the Istio CNI enabled so it's sounding like a common setup issue or common redirect problem.  

@alunsford  -- could you clarify how you determine the traffic is not being redirected?

**Comment by alunsford** [comment]
@tiswanso in 1.0.6 we are able to tail the istio-proxy sidecar logs and see entries as it receives traffic. We tried setting different log levels for the sidecar in 1.1.1 (trace, debug, info, warn) and did not see entries. Additionally, we have kialia/jaeger in the cluster and they were not showing traffic flowing through the mesh.

**Comment by alunsford** [comment]
So today we tried upgrading from 1.0.6 to 1.1.7 and the process worked with a few tweaks to the included helm chart. Lots of changes went in between 1.1.1 and 1.1.7 so I can't say which one in particular addressed the issue we were facing.

Previously, we were unable to traffic in Kiali, nor was it being reported by prometheus. Now we see it in both prometheus and kiali. Envoy access logs were disabled by default at some point so after enabling them via the helm chart, we were able to see logs from the istio-proxy sidecar :+1:.
